ports/psoc6/modules/machine/modmachine.h

```diff
@@ -1,21 +1,30 @@
 #ifndef MICROPY_INCLUDED_PSOC6_MODMACHINE_H
 #define MICROPY_INCLUDED_PSOC6_MODMACHINE_H
 
-#include "py/obj.h"
+// std includes
+#include <stdio.h>
 
+// mpy includes
+#include "py/obj.h"
+#include "py/gc.h"
+#include "py/runtime.h"
+#include "py/mphal.h"
+#include "shared/runtime/pyexec.h"
+#include "extmod/machine_bitstream.h"
+#include "extmod/machine_i2c.h"
+#include "extmod/machine_mem.h"
+#include "extmod/machine_pulse.h"
+#include "extmod/machine_pwm.h"
+#include "extmod/machine_signal.h"
+#include "extmod/machine_spi.h"
+
+// port-specific includes
+#include "drivers/psoc6_system.h"
 
 extern const mp_obj_type_t machine_pin_type;
 
-
-void machine_pin_init(void);
```

Comment:
These methods are no longer visible to the outside world if not declared here. So they should be listed here.

Reply:
Almost all functions (except these two) in the `modmachine.c` are of `static` nature. It is quite well adopted across all the ports and most of the MPY files (py/*). Hence these cannot be redeclared in any other file (eg. in the `modmachine.h` header file). 

In case we want to make these functions (or some of them) public -> accessible from other files, then we have to remove their `static` linkages. Please let me know if my thinking is correct.

Reply:
There are 2 kinds of init function used in MPY : some are called in main() in order to generally prepare the tool for using a certain module. They have to be non-static. Then there are some that are called as the user calls the constructor or triggers some other action. Those are normally object specific and will potentially be called many times whereas the first type is called only once. 
My intention for machine_pin_init() was the first type, i.e. called once in main.

Reply:
Done as requested for these two functions (init and deinit) which may be called from outside (from main.c).